std-bits/table/src/main/java/org/enso/table/data/column/builder/string/StringStorageBuilder.java

```diff
@@ -1,12 +1,11 @@
 package org.enso.table.data.column.builder.string;
 
-import org.enso.table.data.column.storage.Storage;
 import org.enso.table.data.column.storage.StringStorage;
 
 /** A column builder appending all the values passed to it in an unchanged form. */
 public class StringStorageBuilder extends StorageBuilder {
 
-  private Object[] data;
+  private String[] data;
```

Comment:
Is us doing this better than just using an `ArrayList<T>` feels like we are doing this for little gain once especially we have the new polyglot arrays

Reply:
Hmm, well, to some extent. I'd argue that our implementation is still more specialized. It avoids copying (whereas `ArrayList<T>::toArray` would copy IIRC. And it has the retype logic which we would still need to add.

So I'd be for keeping it as-is, although you are right that there is quite a lot of similarity.